linera-service/src/storage.rs

```diff
@@ -278,64 +304,70 @@ impl StoreConfig {
-            StoreConfig::RocksDb(config) => Ok(RocksDbStore::test_existence(config).await?),
+            StoreConfig::RocksDb(config, namespace) => {
+                Ok(RocksDbStore::exists(&config, &namespace).await?)
+            }
             #[cfg(feature = "aws")]
-            StoreConfig::DynamoDb(config) => Ok(DynamoDbStore::test_existence(config).await?),
+            StoreConfig::DynamoDb(config, namespace) => {
+                Ok(DynamoDbStore::exists(&config, &namespace).await?)
+            }
             #[cfg(feature = "scylladb")]
-            StoreConfig::ScyllaDb(config) => Ok(ScyllaDbStore::test_existence(config).await?),
+            StoreConfig::ScyllaDb(config, namespace) => {
+                Ok(ScyllaDbStore::exists(&config, &namespace).await?)
+            }
         }
     }
 
     /// Deletes only one table of the database
     pub async fn initialize(self) -> Result<(), ViewError> {
         match self {
-            StoreConfig::Memory(_) => Err(ViewError::ContextError {
+            StoreConfig::Memory(_, _) => Err(ViewError::ContextError {
                 backend: "memory".to_string(),
-                error: "delete_single does not make sense for memory storage".to_string(),
+                error: "initialize does not make sense for memory storage".to_string(),
             }),
             #[cfg(feature = "rocksdb")]
-            StoreConfig::RocksDb(config) => {
-                RocksDbStore::initialize(config).await?;
+            StoreConfig::RocksDb(config, namespace) => {
+                RocksDbStore::maybe_create_and_connect(&config, &namespace).await?;
                 Ok(())
             }
             #[cfg(feature = "aws")]
-            StoreConfig::DynamoDb(config) => {
-                DynamoDbStore::initialize(config).await?;
+            StoreConfig::DynamoDb(config, namespace) => {
+                DynamoDbStore::maybe_create_and_connect(&config, &namespace).await?;
                 Ok(())
             }
             #[cfg(feature = "scylladb")]
-            StoreConfig::ScyllaDb(config) => {
-                ScyllaDbStore::initialize(config).await?;
+            StoreConfig::ScyllaDb(config, namespace) => {
+                ScyllaDbStore::maybe_create_and_connect(&config, &namespace).await?;
                 Ok(())
             }
         }
     }
 
-    /// List all the tables of the database
-    pub async fn list_tables(self) -> Result<Vec<String>, ViewError> {
+    /// List all the namespaces of the storage
+    pub async fn list_all(self) -> Result<Vec<String>, ViewError> {
         match self {
-            StoreConfig::Memory(_) => Err(ViewError::ContextError {
+            StoreConfig::Memory(_, _) => Err(ViewError::ContextError {
                 backend: "memory".to_string(),
-                error: "list_tables is not supported for the memory storage".to_string(),
+                error: "list_all is not supported for the memory storage".to_string(),
```

Comment:
Well, no.
I would prefer "does not make sense". This is because whenever we create a memory instance it is specific to the scope being considered. And so, a list_all could only return 0 entries, since it is about the memory instances just created.